cmd/admin-rpc-client.go

```diff
@@ -135,6 +142,77 @@ func (rc remoteAdminClient) GetConfig() ([]byte, error) {
 	return reply.Config, nil
 }
 
+// WriteTmpConfig - writes config file content to a temporary file on
+// the local server.
+func (lc localAdminClient) WriteTmpConfig(tmpFileName string, configBytes []byte) (int64, error) {
+	tmpFile, err := os.Create(filepath.Join(os.TempDir(), tmpFileName))
+	if err != nil {
+		errorIf(err, "Failed to create temporary config file.")
+		return 0, err
+	}
+
+	defer tmpFile.Close()
+
+	reader := bytes.NewReader(configBytes)
+	n, err := io.Copy(tmpFile, reader)
+	if err != nil {
+		errorIf(err, "Failed to write into temporary config file.")
+		return 0, err
+	}
+
+	return n, nil
+}
+
+// WriteTmpConfig - writes config file content to a temporary file on
+// a remote node.
+func (rc remoteAdminClient) WriteTmpConfig(tmpFileName string, configBytes []byte) (int64, error) {
+	wArgs := WriteConfigArgs{
+		TmpFileName: tmpFileName,
+		Buf:         configBytes,
+	}
+
+	wReply := WriteConfigReply{}
+
+	err := rc.Call("Admin.WriteTmpConfig", &wArgs, &wReply)
+	if err != nil {
+		errorIf(err, "Failed to write temporary config file.")
+		return 0, err
+	}
+
+	return wReply.N, nil
+}
+
+func (lc localAdminClient) CommitConfig(tmpFileName string) error {
+	configDir, err := getConfigPath()
+	if err != nil {
+		errorIf(err, "Failed to get config path.")
+		return err
+	}
+
+	configFilePath := filepath.Join(configDir, "config.json")
```

Comment:
Yes, you are right. I am making a change, as we speak, to write the temporary config file inside the config dir and not os.TempDir() as it is now. When I tested locally on a distributed Minio test setup, `/tmp/` is being shared by all minio server instances running on the node. Config directory OTOH could be provided separate for each instance.